pkg/reconciler/events/auditlogs/auditlogs.go

```diff
@@ -64,6 +70,28 @@ func (c *Reconciler) ReconcileKind(ctx context.Context, s *v1alpha1.CloudAuditLo
 	s.Status.InitializeConditions()
 	s.Status.ObservedGeneration = s.Generation
 
+	// If GCP ServiceAccount is provided, configure workload identity.
+	if s.Spec.ServiceAccount != nil {
+		// Create corresponding k8s ServiceAccount if doesn't exist, and add ownerReference to it.
+		kServiceAccountName := psresources.GenerateServiceAccountName(s.Spec.ServiceAccount)
+		kServiceAccount, err := c.serviceAccountLister.ServiceAccounts(s.Namespace).Get(kServiceAccountName)
+		if err != nil {
+			if !apierrs.IsNotFound(err) {
+				logging.FromContext(ctx).Desugar().Error("Failed to get k8s service account", zap.Error(err))
+				return reconciler.NewEvent(corev1.EventTypeWarning, workloadIdentityFailedReason, "Getting k8s service account failed with: %s", err)
+			}
+			kServiceAccount = nil
+		}
+		kServiceAccount, event := c.CreateServiceAccount(ctx, s, kServiceAccount)
```

Comment:
change event --> err, otherwise it sounds weird

Reply:
I assume it will return a reconciler.Event once we finish the migration of generated reconciler for pubsub base reconciler. But you are right, event sounds weird...